### ISSUE TITLE
Implicit cast of source type Double to narrower destination type int.

### ISSUE THREAD

**Issue by axkr** [jgrapht/jgrapht#1087]
See:
- https://github.com/jgrapht/jgrapht/blob/6aba8e81053660997fe681c50974c07e312027d1/jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/CapacitatedSpanningTreeAlgorithm.java#L155

Should `currentCapacity ` be defined as `double`, because the `demands` map returns `Double` values?
```
double currentCapacity = 0;
```

**Comment by jsichi** [comment]
@christophgruene do you agree?

**Comment by christophgruene** [comment]
Yes, I agree. It makes no sense to cast from Double to int to compare it with a double in this situation.
